src/modules/ipops/ipops_mod.c

```diff
@@ -1292,6 +1297,31 @@ static int ki_dns_query(sip_msg_t *msg, str *naptrname, str *pvid)
 	return dns_update_pv(naptrname, pvid);
 }
 
+/**
+ *
+ */
+static int w_ptr_query(sip_msg_t *msg, char *ip, char *pv_name)
+{
+	str ip_address;
+	str name;
+
+	if(msg == NULL) {
+		LM_ERR("received null msg\n");
+		return -1;
+	}
+
+	if(fixup_get_svalue(msg, (gparam_t *)ip, &ip_address) < 0) {
+		LM_ERR("cannot get the IP address\n");
+		return -1;
```

Comment:
Just thinking loud - any reason you want different returns in different situations in your scripts? All errors return exactly the same value here.

Reply:
I followed the `w_dns_query` as a base and assumed this is how error reporting is done. How would you like me to change them?

Reply:
Depends if you think there's any point of separating the various errors from each other in the script of if it doesn't matter - the coder just want to know that it failed, not caring about how. There are several functions in other modules, like TM, that has multiple return values. It's really up to you, just a suggestion that you consider this.

Reply:
Main reason i see for separating errors is for debugging purposes. 

Another thing, that might be good, would be to separate the failure over bad input and successful call with no hostname associated ([getnameinfo](https://man7.org/linux/man-pages/man3/getnameinfo.3.html)). Do you think this makes sense to separate and document it?

Other than that, as you have said the code just want to know it had failed and not how.

Reply:
Seems like you have reasons for multiple return values, so go ahead.